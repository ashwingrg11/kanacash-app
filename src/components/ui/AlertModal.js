import React from 'react';
import {
  View,
  StyleSheet,
  Modal,
  TouchableWithoutFeedback,
  TouchableOpacity,
} from 'react-native';
import {SemiBoldText, RegularText} from '~/components/ui/Text';
import theme from '~/components/theme/Style';

export default function AlertModal({
  visible,
  onRequestClose,
  onPressOk,
  title,
  message,
  closeButton,
}) {
  return (
    <Modal
      statusBarTranslucent={true}
      animationType="fade"
      transparent={true}
      onRequestClose={onRequestClose}
      visible={visible}>
      <TouchableWithoutFeedback style={styles.mainView}>
        <View style={styles.mainView}>
          <View style={styles.messageWrapper}>
            <View style={styles.alertMessage}>
              {title && <SemiBoldText text={title} invert />}
              <RegularText text={message} style={styles.alertMessageText} />
            </View>

            {!closeButton ? (
              <View style={styles.rowGroup}>
                <TouchableOpacity
                  onPress={onPressOk}
                  style={styles.singleBtnView}>
                  <SemiBoldText text="Yes" invert />
                </TouchableOpacity>

                <TouchableOpacity
                  onPress={onRequestClose}
                  style={styles.cancelButton}>
                  <SemiBoldText text="No" />
                </TouchableOpacity>
              </View>
            ) : (
              <TouchableOpacity
                onPress={onPressOk}
                style={styles.singleBtnView}>
                <SemiBoldText text={closeButton} invert />
              </TouchableOpacity>
            )}
          </View>
        </View>
      </TouchableWithoutFeedback>
    </Modal>
  );
}

const styles = StyleSheet.create({
  mainView: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(0,0,0,0.44)',
  },
  alertMessage: {
    // paddingLeft: 20,
    // paddingRight: 20,
    marginBottom: 20,
    alignItems: 'center',
    justifyContent: 'center',
  },
  messageWrapper: {
    width: '80%',
    backgroundColor: '#1F1F1F',

    padding: 20,
    borderRadius: 5,
  },
  alertMessageText: {
    marginTop: 10,
    fontSize: 16,
    color: '#fff',
    textAlign: 'center',
  },
  cancelButton: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: 2,
    paddingHorizontal: 40,
    backgroundColor: '#fff',
    borderRadius: 50,
  },
  singleBtnView: {
    justifyContent: 'center',
    alignItems: 'center',
    paddingVertical: 2,
    paddingHorizontal: 40,
    backgroundColor: theme.red,
    borderColor: theme.red,
    borderRadius: 50,
  },
  rowGroup: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
});
