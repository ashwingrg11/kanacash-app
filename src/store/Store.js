import {applyMiddleware, createStore} from 'redux';
import {persistReducer, persistStore} from 'redux-persist';
import thunk from 'redux-thunk';
import rootReducer from './rootReducer';
import {composeWithDevTools} from 'redux-devtools-extension';
import AsyncStorage from '@react-native-community/async-storage';

const config = {
  key: 'root',
  storage: AsyncStorage,
  whitelist: [],
  blackList: [],
};

const reducer = persistReducer(config, rootReducer);
const middleware = composeWithDevTools(applyMiddleware(thunk));

export default data => {
  const store = createStore(reducer, middleware);
  const persistor = persistStore(store);
  return {persistor, store};
};
