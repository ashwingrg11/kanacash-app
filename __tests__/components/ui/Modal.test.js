import 'react-native';
import React from 'react';

import {SafeAreaProvider} from 'react-native-safe-area-context';

import AlertModal from '~/components/ui/Modal';
import Store from '~/store/Store';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

describe('<Modal/>', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <SafeAreaProvider store={Store}>
          <AlertModal />
        </SafeAreaProvider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
