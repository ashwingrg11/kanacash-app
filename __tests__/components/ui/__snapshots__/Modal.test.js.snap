// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal/> renders correctly 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
/>
`;
